launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/ContextDataManager.java

```diff
@@ -133,26 +133,27 @@ private void initDataInternal(
 
             oldData = flags;
             flags = newData;
-            if (index == null) {
-                index = environmentStore.getIndex();
-            }
-            newIndex = index.updateTimestamp(contextId, System.currentTimeMillis())
-                    .prune(maxCachedContexts, removedContextIds);
-            index = newIndex;
 
-            for (String removedContextId: removedContextIds) {
-                environmentStore.removeContextData(removedContextId);
-                logger.debug("Removed flag data for context {} from persistent store", removedContextId);
-            }
-            if (writeFlagsToPersistentStore && maxCachedContexts != 0) {
-                environmentStore.setContextData(contextId, newData);
+            if (writeFlagsToPersistentStore) {
+                List<String> removedContextIds = new ArrayList<>();
+                newIndex = index.updateTimestamp(contextId, System.currentTimeMillis())
+                        .prune(maxCachedContexts, removedContextIds);
+                index = newIndex;
+
+                for (String removedContextId: removedContextIds) {
+                    environmentStore.removeContextData(removedContextId);
+                    logger.debug("Removed flag data for context {} from persistent store", removedContextId);
+                }
+
+                environmentStore.setContextData(contextId, fingerprint, newData);
+                environmentStore.setIndex(newIndex);
+
+                if (logger.isEnabled(LDLogLevel.DEBUG)) {
+                    logger.debug("Stored context index is now: {}", newIndex.toJson());
+                }
+
                 logger.debug("Updated flag data for context {} in persistent store", contextId);
             }
-            environmentStore.setIndex(newIndex);
-        }
-
-        if (logger.isEnabled(LDLogLevel.DEBUG)) {
-            logger.debug("Stored context index is now: {}", newIndex.toJson());
```

Comment:
For reviewers: Android was updating the index timestamp when switching context.  This is not consistent with JS, Flutter, and the iOS SDK.  Updated this logic so that the index is only updated when new flag data needs to be written to persistence.